semantics/src/main/scala/aqua/semantics/rules/types/TypesAlgebra.scala

```diff
@@ -41,4 +41,10 @@ trait TypesAlgebra[S[_], Alg[_]] {
 
   // End scope; if return was expected but not checked, fail
   def endArrowScope(token: Token[S]): Alg[List[ValueRaw]]
+
+  def checkTypeCompatibility(
```

Comment:
Both type and semantic of this method is exactly same as `ensureTypeMatches`: you expect `type`, you give `valueType`, etc.

Why another method for the same problem?